File: features/master/src/main/java/com/nivasafinance/features/master/codemaster/dto/MasterCodeWithValuesResponse.java
package com.nivasafinance.features.master.codemaster.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;
import java.util.Map;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class MasterCodeWithValuesResponse {
    private Long id;
    private String key;
    private String name;
    private String description;
    private Map<String, String> nameMap;
    private Map<String, String> descriptionMap;
    private Boolean isSystemDefined;
    private Long parentId;
    private List<CodeValueResponse> values;
}

File: features/master/src/main/java/com/nivasafinance/features/master/codemaster/dto/MasterCodeWithValuesResponse.kt
package com.nivasafinance.features.master.codemaster.dto

import java.util.UUID

data class MasterCodeWithValuesResponse(
    val id: UUID?,
    val key: String,
    val name: String,
    val description: String?,
    val isSystemDefined: Boolean,
    val parentId: UUID?,
    val values: List<CodeValueResponse>
)
